src/components/entry/EntryDrawer.js

```diff
@@ -0,0 +1,260 @@
+import { useWindowSize } from '@reach/window-size'
+import { ArrowBack as ArrowBackIcon } from '@styled-icons/boxicons-regular'
+import { Pencil as PencilIcon } from '@styled-icons/boxicons-solid'
+import { useEffect, useRef, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import styled from 'styled-components/macro'
+
+import IconButton from '../ui/IconButton'
+import Card from './Card'
+import Entry from './Entry'
+import EntryImagesCard from './EntryImagesCard'
+
+const ENTRY_IMAGE_HEIGHT = 250
+
+const INITIAL_IMAGE_HEIGHT_SCALAR = 0.6
+
+const FOOTER_HEIGHT = 70
+
+const BUTTON_HEIGHT = 80
+
+const Container = styled.div`
+  position: absolute;
+  height: 100%;
+  overflow: hidden;
+  width: 100%;
+  bottom: 70px;
+  left: 0;
+
+  .pane {
+    background: none;
+    ${({ isFullScreen }) => isFullScreen && `padding-top: 0;`}
+    ${({ showEntryImages, isFullScreen }) =>
+      !showEntryImages && isFullScreen && `box-shadow: none;`}
+  }
+
+  .draggable {
+    background: white;
+    z-index: -1;
+    padding-top: 10px;
+    ${({ showEntryImages }) => !showEntryImages && `border-radius: 13px;`}
+  }
+
+  .move {
+    ${({ isFullScreen }) => isFullScreen && `visibility: hidden;`}
+  }
+
+  .entry-main-card {
+    background: white;
+    height: 100% !important;
+  }
+`
+
+const EntryImages = styled.div`
+  width: 100%;
+  height: ${ENTRY_IMAGE_HEIGHT}px;
+  position: absolute;
+  top: 0;
+  transform: translateY(
+    ${({ heightScalar }) => -heightScalar * ENTRY_IMAGE_HEIGHT}px
+  );
+  transition: transform 0.15s linear;
+  z-index: -10;
+`
+
+const Buttons = styled.div`
+  position: absolute;
+  left: 0;
+  right: 0;
+  top: 0;
+  z-index: 12;
+  padding: 16px;
+  display: flex;
+  justify-content: space-between;
+`
+
+const EntryButton = styled(IconButton)`
+  background-color: rgba(0, 0, 0, 0.45);
+  border: none;
+  svg {
+    color: white;
+  }
+`
+
+const Backdrop = styled.div`
+  width: 100%;
+  height: ${BUTTON_HEIGHT}px;
+  position: absolute;
+  top: 0;
+  background: white;
+  z-index: -10;
+  transform: ${({ isFullScreen }) =>
+    isFullScreen ? `translateY(${-BUTTON_HEIGHT}px);` : `translateY(10px);`};
+  transition: transform 0.15s linear;
+`
+
+const EntryDrawer = ({
+  locationData,
+  reviews,
+  showEntryImages,
+  isLoading,
+  entryOverview,
+  entryReviews,
+}) => {
+  const history = useHistory()
+  const cardRef = useRef()
+  const [drawer, setDrawer] = useState()
+  const [isFullScreen, setIsFullScreen] = useState(false)
+
+  const { height: windowHeight } = useWindowSize()
```

Comment:
Let's add a TODO / note here that resizing your screen will break the drawer. People's screen sizes shouldn't be changing often (especially if we enforce screenlock), but its worth noting